pybossa/api/task_run.py

```diff
@@ -68,6 +68,7 @@ class TaskRunAPI(APIBase):
     def _preprocess_post_data(self, data):
         with_encryption = app.config.get('ENABLE_ENCRYPTION')
         upload_root_dir = app.config.get('S3_UPLOAD_DIRECTORY')
+        conn_name = "S3_TASKRUN_V2" if app.config.get("S3_CONN_TYPE_V2") else "S3_TASKRUN"
```

Comment:
Consider defining the related variables in the class `TaskRunAPI` so all methods have access to it from a single location.

```python
class TaskRunAPI(APIBase):
   conn_name = "S3_TASKRUN_V2" if app.config.get("S3_CONN_TYPE_V2") else "S3_TASKRUN"
   bucket = app.config.get("S3_BUCKET_V2") if app.config.get("S3_CONN_TYPE_V2") else app.config.get("S3_BUCKET")
```

or

```python
class TaskRunAPI(APIBase):
   version_str = "_V2" if app.config.get("S3_CONN_TYPE_V2") else ""
   conn_name = "S3_TASKRUN" + version_str
   bucket_name = "S3_BUCKET" + version_str
   bucket = app.config.get(bucket_name)
```

Reply:
Make sense. I'll update the code. Thanks!